sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -498,6 +499,117 @@ class Analyzer(
     }
   }
 
+  val ResolveOutputColumns = new ResolveOutputColumns
+  class ResolveOutputColumns extends Rule[LogicalPlan] {
```

Comment:
I'll fix this and the `protected` method. I think I was instantiating this rule multiple times at one point and this is left-over.